### DISCUSSION TITLE
Have graphs of position of the drones

### DISCUSSION THREAD

**Discussion by ghost** [USC-ACTLab/crazyswarm#783] [Q&A]
Hello,

Is there a way to have graphs of the position/velocity of the swarm/drones after a flight ? I think it is doable with the position() command of the API, I am juste asking if there is already an automated way to do that.

Thanks a lot

**Comment by whoenig** [comment]
You can write csv files at https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/hover_swarm.launch#L41.
Generally, I'd recommend using rosbags and then write a script that reads your rosbag and generates whatever plots you desire from it.

**Reply by ghost** [reply]
Hi,

I have tried to turned the flag at true in the launch file (https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/hover_swarm.launch#L41) but when I do a flight I don't know how to activate the writing of those CSV and I don't know where they are stored.
Also I have tried to record the position (x, y, z in the motion capture frame) of my drone using a ROS bag, to do so I have recorded the /tf
But I am struggling to extract data from my recording : even after doing internet researches I didn't manage to extract the x, y and z evolution of my drone from the tf2 messages in a python script (I don't know to call the geometry message part of the tf2, and after extract data from this geometry message part).

These may be basics questions, but if someone has an answer, i would love to hear it !

Best regards
